### DISCUSSION TITLE
Search function to look in title, content, tags and categories

### DISCUSSION THREAD

**Discussion by kmunkholm** [adityatelange/hugo-PaperMod#141] [Q&A] [answered]
Hi,

I am so grateful for this theme - it is everything I could possibly ask for!

Among the many things I really like about this theme is the **search functionality**. However, it seems to me that it only searches the titles of posts - not the content, categories or tags, which I would like it to. I have serched the [Fuse.js](https://fusejs.io) help sections and the PaperMod theme documentation, but have not been able to solve the issue. It is quite possible that I misunderstand the inner workings of Fuse.js, but it appears to me that it should be possible to search all the content of posts. 

This is in my config.toml:
```
  [params.fuseOpts]
  isCaseSensitive = false
  shouldSort = true
  location = 0
  distance = 1000
  threshold = 0.4
  minMatchCharLength = 0
  keys = [ "title", "tags", "categories", "permalink", "summary", "content" ] 
```
Any help will be greatly appreciated (and apologies in advance if I have overlooked the obvious)!

**Comment by adityatelange** [comment, marked as answer]
@kmunkholm Take a look here 
https://github.com/adityatelange/hugo-PaperMod/blob/6c83df566c2a8be51985808fd1fd18bdc84e7d00/layouts/_default/index.json#L1-L6
This is the index where `search` searches for into,
namely
- title
- content
- url
- summary

It does search in content, although
It does not search in Taxonomies i.e Tags / Categories.
keys used can be some, all or none from `["title", "permalink", "summary", "content"]` , but cannot be more unless you ovveride `index.json`

refer commit message of 77ff1e6b452ad64eeee5178667827323d0a1644d

**Reply by kmunkholm** [reply]
Thank you so much, @adityatelange for the thorough reply. It explains why it does not search Taxonomies, which is actually perfectly fine with me as long as it searches the content of posts. But I still cannot get that to work. 

I took a look at the [PaperMod sample site](https://adityatelange.github.io/hugo-PaperMod/) and it also does not seem to be searching the content of the posts. This could be because the params for the site are not what I think they are, but I am guessing that they are identical to those specified [here](https://adityatelange.github.io/hugo-PaperMod/posts/papermod/papermod-features/#search), which means the content should be searched. I could very well be wrong, but it does not seem so. For example, a search for the words "version" or "create", even though they appear under the first headline of the [PaperMod Installation post](https://adityatelange.github.io/hugo-PaperMod/posts/papermod/papermod-installation/), comes up empty. Conversely, it detects the word "update", also found in the above post, but not the word "updating", which also appears in the post. There is no doubt that the titles of posts are searched. Could it be that the other keys (or at least "content") in the Params for Fuse.js is not capturing where search is looking?

**Comment by kmunkholm** [comment]
Unfortunately I have no programming expertise so I am sorry I am not able to be of any help in resolving this issue. I hope @adityatelange or someone with experience in implementing Fuse.js in HUGO will have the chance to look into this.

If it can be of any help, many people seem to have found the following helpful in implementing Fuse.js search to HUGO sites:
https://gist.github.com/eddiewebb/735feb48f50f0ddd65ae5606a1cb41ae
https://gist.github.com/cmod/5410eae147e4318164258742dd053993
https://github.com/kaushalmodi/hugo-search-fuse-js
An example of a HUGO site with functioning Fuse.js search implented using the last link above is: https://live.cogley.info.

**Reply by adityatelange** [reply]
I'll look into it, thanks

**Reply by adityatelange** [reply]
@kmunkholm f64c44c8d2a19fd0ad833bea92033b0b13e3c276 should fix the issue you were facing :)

**Reply by kmunkholm** [reply]
Thank you, @adityatelange! It works like a charm - searches everywhere in posts. Just was I was looking for! 
Thank you once again for a great theme!
